programs/drift/src/controller/liquidation.rs

```diff
@@ -969,13 +967,28 @@ pub fn liquidate_perp_with_fill(
     )?;
 
     let existing_direction = user.perp_positions[position_index].get_direction();
+    let max_liquidation_fee =
```

Comment:
1) should use safe_math
2) should prob just call get_ref once and then drop it

could potentially also just make this a method on the perp market account?